### PR TITLE
#164798035 users should be able to make a request to become a reviewer

### PR DESCRIPTION
**Pull request by amaechi-chuks**
#### Description
Currently, admin cannot get all users that made a request to become a reviewer. Users cannot be able to request to become a reviewer and also remove his/her request. This PR gives users access to request to become a reviewer, also helps an admin to get all users that requested to to become a reviewer.
#### Type of change

- [x] New Feature(admin get all user request)
- [x] New Feature( user request to become a reviewer)
#### How Has This Been Tested?

- [x] End to End testing

#### Checklist:
- [x] design and setup database table for user request and define the relationship with the user table
- [x] create get all users controller
- [x] create make a request controller
- [x] create remove your request controller
- [x] write test cases

#### PT-ID
[164798037](https://www.pivotaltracker.com/n/projects/2318856/stories/164798037)
#### Screenshots
**Admin get all users request**
image
**User make a request**
image
**user remove request**
image

#### Questions:
N/A

### CODE REVIEW

File: server/controllers/admin.controllers.js [hunk 1]
```diff
@@ -89,5 +89,44 @@ const deactivateReviewer = async (req, res) => {
   }
 };
 
-const upgradeDegradeUserReviewer = { activateReviewer, deactivateReviewer };
+const getAllReviewerRequest = async (req, res) => {
```
Comment (rotimi-babalola) [inline server/controllers/admin.controllers.js]: Should be `getAllReviewerRequests`

File: server/controllers/user-request.controllers.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        },
+      });
+    }
+    const requested = await Request.create({
+      user_id: userId,
+      is_reviewer: false,
+      is_reported: false,
+    });
+    return res.status(201).json({
+      message: 'Successfully make a request',
```
Comment (rotimi-babalola) [inline server/controllers/user-request.controllers.js]: Should be `You have successfully made a request`

File: server/controllers/user-request.controllers.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    const requested = await Request.destroy({
+      where: {
+        user_id: userId,
+        is_reviewer: false,
+        is_reported: false,
+      },
+    });
+    if (requested) {
+      return res.status(200).json({
+        message: 'Successfully unchecked request',
```
Comment (rotimi-babalola) [inline server/controllers/user-request.controllers.js]: Do the same thing like I stated earlier - Use syntactically correct sentences

File: server/middlewares/user-request.middleware.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+      user_id: userId,
+      is_reviewer: false,
+      is_reported: false,
+    },
+  });
+  if (findOneRequest) {
+    return res.status(403).json({
+      status: 403,
+      errors: {
+        body: ['You have already make a request'],
```
Comment (rotimi-babalola) [inline server/middlewares/user-request.middleware.js]: Should be `... made ... `

File: server/middlewares/user-request.middleware.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+      user_id: userId,
+      is_reviewer: false,
+      is_reported: false,
+    },
+  });
+  if (!findOneRequest) {
+    return res.status(403).json({
+      status: 403,
+      errors: {
+        body: ['You have already unchecked your request'],
```
Comment (rotimi-babalola) [inline server/middlewares/user-request.middleware.js:37]: 👍

File: server/routes/user-request.routes.js [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+const router = express.Router();
+
+/**
+ * @swagger
+ *
+ * /request:
+ *   post:
+ *     tags:
+ *       - request
+ *     description: authenticated user can make a request to becaome a reviewer
```
Comment (rotimi-babalola) [inline server/routes/user-request.routes.js]: Should be `become`

File: server/controllers/admin.controllers.js [hunk 7]
```diff
@@ -89,5 +89,44 @@ const deactivateReviewer = async (req, res) => {
...
+    }
+    const getAllUserRequest = await Request.findAll({
+      where: {
+        is_reviewer: false,
+        is_reported: false,
+      },
+    });
+    if (getAllUserRequest) {
+      return res.status(200).json({
+        message: 'Here is a list of users request',
```
Comment (rotimi-babalola) [inline server/controllers/admin.controllers.js]: I think this message can be better

File: server/controllers/admin.controllers.js (hunk 1 above)
Comment (amaechi-chuks) [reply server/controllers/admin.controllers.js]: Noted and fixed

File: server/controllers/admin.controllers.js (hunk 7 above)
Comment (amaechi-chuks) [reply server/controllers/admin.controllers.js]: Noted and fixed

File: server/controllers/user-request.controllers.js (hunk 2 above)
Comment (amaechi-chuks) [reply server/controllers/user-request.controllers.js]: Noted and fixed

File: server/routes/user-request.routes.js (hunk 6 above)
Comment (amaechi-chuks) [reply server/routes/user-request.routes.js]: Noted and fixed

File: server/controllers/user-request.controllers.js (hunk 3 above)
Comment (amaechi-chuks) [reply server/controllers/user-request.controllers.js]: Noted and fixed

File: server/middlewares/user-request.middleware.js (hunk 4 above)
Comment (amaechi-chuks) [reply server/middlewares/user-request.middleware.js]: Noted and fixed

File: server/middlewares/user-request.middleware.js (hunk 5 above)
Comment (amaechi-chuks) [reply server/middlewares/user-request.middleware.js:37]: Thank you.

File: server/controllers/admin.controllers.js [hunk 8]
```diff
@@ -106,16 +106,16 @@ const getAllReviewerRequest = async (req, res) => {
...
         is_reviewer: false,
         is_reported: false,
       },
     });
-    if (getAllUserRequest) {
+    if (getAllUsersRequest) {
       return res.status(200).json({
-        message: 'Here is a list of users request',
-        allUsersRequest: getAllUserRequest,
+        message: 'All users request',
```
Comment (rotimi-babalola) [inline server/controllers/admin.controllers.js]: We can have - `List of users requesting to be reviewers`

**Pull request merged**